src/main/__tests__/helpers/urlParser.spec.ts

```diff
@@ -0,0 +1,115 @@
+import createEventManager from '../../identityEventManager'
+import createUrlParser from '../../urlParser'
+import { delay } from '../../../utils/promise'
+
+function createServices() {
+  const eventManager = createEventManager()
+  const urlParser = createUrlParser(eventManager)
+
+  return { eventManager, urlParser }
+}
+
+describe('parseUrlFragment', () => {
+  test('with success url', async () => {
+    expect.assertions(3)
+
+    // Given
+    const { eventManager, urlParser } = createServices()
+
+    const idToken =
+      'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIn0.Pd6t82tPL3EZdkeYxw_DV2KimE1U2FvuLHmfR_mimJ5US3JFU4J2Gd94O7rwpSTGN1B9h-_lsTebo4ua4xHsTtmczZ9xa8a_kWKaSkqFjNFaFp6zcoD6ivCu03SlRqsQzSRHXo6TKbnqOt9D6Y2rNa3C4igSwoS0jUE4BgpXbc0'
+    const accessToken = 'kjbsdfljndvlksndfv'
+    const expiresIn = 1800
+    const tokenType = 'Bearer'
+
+    const authenticatedHandler = jest.fn().mockName('authenticatedHandler')
+    eventManager.on('authenticated', authenticatedHandler)
+
+    const authenticationFailedHandler = jest.fn().mockName('authenticationFailedHandler')
+    eventManager.on('authentication_failed', authenticationFailedHandler)
+
+    // When
+    const result = urlParser.checkUrlFragment(
+      'https://example.com/login/callback#' +
+        [
+          `id_token=${idToken}`,
+          `access_token=${accessToken}`,
+          `expires_in=${expiresIn}`,
+          `token_type=${tokenType}`
+        ].join('&')
+    )
```

Comment:
Why using fragment params here? Ik those are kinda "hidden" params but maybe you can add more 🙂

Reply:
This is moved coded only: it tests fragment parsing 😄